### PR TITLE
Introduce EDispKernelMap for SpectrumDataset

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request introduces the `EDispKernelMap` in the `SpectrumDataset`. For the moment the stacking still relies on the `IRFStacker`. Removing it will be the subject of another PR.

**Dear reviewer**

### CODE REVIEW

File: gammapy/datasets/tests/test_spectrum.py [hunk 1]
```diff
@@ -412,7 +412,8 @@ def test_to_from_ogip_files(self, tmp_path):
 
         assert_allclose(self.on_counts.data, newdataset.counts.data)
         assert_allclose(self.off_counts.data, newdataset.counts_off.data)
-        assert_allclose(self.edisp.pdf_matrix, newdataset.edisp.pdf_matrix)
+        print(newdataset.edisp.edisp_map.data)
```
[the remark this answers is not in the document]
Comment (registerrier) [reply gammapy/datasets/tests/test_spectrum.py]: Done

**Comment by registerrier** [comment]
Normally there should be no issues left with the notebooks. I changed the `_edisp_kernel` property not to use the `get_edisp_kernel(position)`, so that spectrum_analysis.ipynb should run.

**Comment by adonath** [comment]
@registerrier Sorry you are right. I looked at an old CI build...

**Comment by adonath** [comment]
Thanks @registerrier!

**Pull request merged**